modules/core/src/main/java/org/apache/ignite/Ignite.java

```diff
@@ -722,4 +723,11 @@ public IgniteLock reentrantLock(String name, boolean failoverSafe, boolean fair,
      * @return {@link DataStorageMetrics} snapshot.
      */
     public DataStorageMetrics dataStorageMetrics();
+
+    /**
+     *
+     * @return
+     */
+    public GridKernalContext context();
```

Comment:
GridKernalContext is internal class and resides in 'internal' package. 
Ignite is a public API interface.
Exposing internal class via public API method doesn't look like a good idea.